src/main/java/appeng/me/cache/GridStorageCache.java

```diff
@@ -84,14 +88,20 @@ public class GridStorageCache implements IStorageGrid {
     private long essentiaTypesTotal;
     private long essentiaTypesUsed;
     private long essentiaCellG;
+    private long essentiaCellB;
     private long essentiaCellO;
     private long essentiaCellR;
     private long essentiaCellCount;
     private int ticksCount;
     private int networkBytesUpdateFrequency;
-    private static final int CELL_RED = 3;
-    private static final int CELL_ORANGE = 2;
+    private final HashMap<ItemStack, Integer> itemCells = new HashMap<>();
+    private final HashMap<ItemStack, Integer> fluidCells = new HashMap<>();
+    private final HashMap<ItemStack, Integer> essentiaCells = new HashMap<>();
```

Comment:
`ItemStack` type is not designed for *hash-based* containers (leke `HashMap` and `HashSet`). So I suggest to use specific container - `ItemStackMap`. Please use implementation of `ItemStackMap` *provided by GTNHLib*.

Reply:
but that will make ae **need** GTNH to run. or just use stack name be the key?